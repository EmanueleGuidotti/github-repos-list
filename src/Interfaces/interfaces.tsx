import CSS from "csstype";
import React from "react";

/**
 * Atomic components interfaces
 */
export interface Link {
  link: string;
  text: string;
}

export interface Item {
  children?: JSX.Element;
  otherStyle?: CSS.Properties;
}

export interface Button {
  text?: string | number;
  onClickCursor?: () => {} | void;
}

export interface GenericInput {
  id: string;
  placeholder: string;
  onKeyUpFunction: (event: React.FormEvent<HTMLInputElement>) => {} | void;
  prefix: JSX.Element;
}

export interface Input {
    search: (value: string) => {} | void;
    id: string;
}

/**
 * Molecular components interfaces
 */
export interface Row {
  totalForks: string;
  totalStars: string;
  link: string;
  name: string;
}

export interface PaginationInterface {
  onClickParent: (page: number | string, pageSize: number | string) => void;
  current: number;
  count: number;
}

/**
 * Pages interfaces
 */
export interface Search {
  search: (value: string) => {} | void;
}

/**
 * Repository interfaces
 */
export interface Repo {
  repo: {
    url: string;
    name: string;
    forks: {
      totalCount: string;
    };
    stargazers: {
      totalCount: string;
    };
  };
}

export interface ReposData {
  search: {
    repositoryCount: number;
    pageInfo: {
      endCursor: string;
      hasNextPage: boolean;
      startCursor: string;
    };
    repos: Repo[];
  };
}

export interface ReposDataTotal {
  repositoryCount: number;
}

export interface ReposDataTotalResult {
  search: ReposDataTotal;
}
