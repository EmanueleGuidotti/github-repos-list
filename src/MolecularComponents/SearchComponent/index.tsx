import InputComponent from "../../AtomicComponents/InputComponent";
import { SearchOutlined } from "@ant-design/icons";
import React from "react";
import { Input } from "../../Interfaces/interfaces";

/**
 * Render a search bar component
 *
 * @prop {Function} search - The search function event
 * @prop {string} id - The id to identify the search bar
 *
 * @returns {JSX.Element}
 */
const SearchComponent = ({ search, id }: Input) => {
  const onKeyUp = (event: React.FormEvent<HTMLInputElement>) => {
    search(event.currentTarget.value);
  };
  return (
    <InputComponent
      id={id}
      placeholder={"Search"}
      onKeyUpFunction={(event: React.FormEvent<HTMLInputElement>) =>
        onKeyUp(event)
      }
      prefix={<SearchOutlined />}
    />
  );
};

export default SearchComponent;
