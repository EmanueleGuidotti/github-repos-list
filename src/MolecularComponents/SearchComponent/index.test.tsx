import React from "react";
import { configure, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
configure({ adapter: new Adapter() });
import toJson from "enzyme-to-json";
import SearchComponent from "./index";
import TestRenderer from "react-test-renderer";
import { Search } from "../../Interfaces/interfaces";

describe("TableRowComponent", () => {
  it("Should render search component", async () => {
    const component = TestRenderer.create(
      <SearchComponent id={"searchComponent"} search={(data) => {}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Should send search event", async () => {
    const mockCallBack = jest.fn();
    const component = mount(
      <SearchComponent
        id={"searchComponent"}
        search={(data: Search) => mockCallBack(data)}
      />
    );
    component.find("span").find("input").simulate("keyUp", { keyCode: 9 });
    expect(mockCallBack).toHaveBeenCalledTimes(1);
    component
      .find("span")
      .find("input")
      .simulate("change", { target: { value: "Hello" } });
    expect(toJson(component)).toMatchSnapshot();
  });
});
