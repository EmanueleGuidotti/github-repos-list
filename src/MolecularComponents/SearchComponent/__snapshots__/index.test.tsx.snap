// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRowComponent Should render search component 1`] = `
<span
  className="ant-input-affix-wrapper"
  onMouseUp={[Function]}
>
  <span
    className="ant-input-prefix"
  >
    <span
      aria-label="search"
      className="anticon anticon-search"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="search"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
        />
      </svg>
    </span>
  </span>
  <input
    className="ant-input"
    id="searchComponent"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    placeholder="Search"
    style={null}
    type="text"
    value=""
  />
</span>
`;

exports[`TableRowComponent Should send search event 1`] = `
<SearchComponent
  id="searchComponent"
  search={[Function]}
>
  <InputComponent
    id="searchComponent"
    onKeyUpFunction={[Function]}
    placeholder="Search"
    prefix={<ForwardRef(SearchOutlined) />}
  >
    <Input
      id="searchComponent"
      onKeyUp={[Function]}
      placeholder="Search"
      prefix={<ForwardRef(SearchOutlined) />}
      type="text"
    >
      <ClearableLabeledInput
        bordered={true}
        element={
          <input
            className="ant-input"
            id="searchComponent"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            placeholder="Search"
            type="text"
          />
        }
        focused={false}
        handleReset={[Function]}
        id="searchComponent"
        inputType="input"
        onKeyUp={[Function]}
        placeholder="Search"
        prefix={<ForwardRef(SearchOutlined) />}
        prefixCls="ant-input"
        triggerFocus={[Function]}
        type="text"
        value="Hello"
      >
        <span
          className="ant-input-affix-wrapper"
          onMouseUp={[Function]}
        >
          <span
            className="ant-input-prefix"
          >
            <ForwardRef(SearchOutlined)>
              <AntdIcon
                icon={
                  Object {
                    "icon": Object {
                      "attrs": Object {
                        "focusable": "false",
                        "viewBox": "64 64 896 896",
                      },
                      "children": Array [
                        Object {
                          "attrs": Object {
                            "d": "M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z",
                          },
                          "tag": "path",
                        },
                      ],
                      "tag": "svg",
                    },
                    "name": "search",
                    "theme": "outlined",
                  }
                }
              >
                <span
                  aria-label="search"
                  className="anticon anticon-search"
                  role="img"
                >
                  <IconReact
                    icon={
                      Object {
                        "icon": Object {
                          "attrs": Object {
                            "focusable": "false",
                            "viewBox": "64 64 896 896",
                          },
                          "children": Array [
                            Object {
                              "attrs": Object {
                                "d": "M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z",
                              },
                              "tag": "path",
                            },
                          ],
                          "tag": "svg",
                        },
                        "name": "search",
                        "theme": "outlined",
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="search"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      key="svg-search"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                        key="svg-search-svg-0"
                      />
                    </svg>
                  </IconReact>
                </span>
              </AntdIcon>
            </ForwardRef(SearchOutlined)>
          </span>
          <input
            className="ant-input"
            id="searchComponent"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            placeholder="Search"
            style={null}
            type="text"
            value="Hello"
          />
        </span>
      </ClearableLabeledInput>
    </Input>
  </InputComponent>
</SearchComponent>
`;
